pkg/machinery/client/client.go

```diff
@@ -302,7 +314,7 @@ func NewClient(cfg *tls.Config, endpoints []string, port int, opts ...grpc.DialO
 	c = &Client{}
 
 	// TODO(smira): endpoints[0] should be replaced with proper load-balancing
-	c.conn, err = grpc.DialContext(context.Background(), fmt.Sprintf("%s:%d", net.FormatAddress(endpoints[0]), port), opts...)
+	c.conn, err = grpc.DialContext(context.Background(), fmt.Sprintf("%s:%d", endpoints[0], port), opts...)
```

Comment:
this became wrong now: it should use `net.JoinHostPort`, but better if we drop the method altogether. It is deprecated for a long time now.